### PR TITLE
GetMapLeafByRevision API

### PR DESCRIPTION
**Pull request by gdbelvin**
Create an API for fetching a single map leaf.
Separating single leaf from batch fetches helps create sensible latency graphs

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by gdbelvin** [comment]
PTAL

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 Not yet released; provisionally v2.0.0 (may change).
 
+### GetMapLeafByRevision API
+New API for fetching a single map leaf separates batch APIs from single shot
```
Comment (Martin2112) [inline CHANGELOG.md]: I think "single shot" is open to being misunderstood. Maybe just remove the last bit and note batch API still exists.

File: server/map_rpc_server.go [hunk 2]
```diff
@@ -112,6 +112,23 @@ func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapL
 	return t.getLeavesByRevision(ctx, req.MapId, req.Index, mostRecentRevision)
 }
 
+// GetLeafByRevision returns an inclusion proof to either the leaf, or nil if the leaf does not exist.
```
Comment (Martin2112) [inline server/map_rpc_server.go]: Can remove "either".

File: server/map_rpc_server.go [hunk 3]
```diff
@@ -112,6 +112,23 @@ func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapL
...
+func (t *TrillianMapServer) GetLeafByRevision(ctx context.Context, req *trillian.GetMapLeafByRevisionRequest) (*trillian.GetMapLeafResponse, error) {
+	ctx, spanEnd := spanFor(ctx, "GetLeafByRevision")
+	defer spanEnd()
+	ret, err := t.getLeavesByRevision(ctx, req.MapId, [][]byte{req.Index}, mostRecentRevision)
+	if err != nil {
+		return nil, err
+	}
+	if got := len(ret.MapLeafInclusion); got != 0 {
```
Comment (Martin2112) [inline server/map_rpc_server.go]: `got != 1` ? A test would have caught this.

Comment (gdbelvin) [reply server/map_rpc_server.go]: Sorry about that.  A gentleman always adds tests.

**Pull request merged**